src/lib/alarmMgr/AlarmManager.cpp

```diff
@@ -0,0 +1,240 @@
+/*
+*
+* Copyright 2015 Telefonica Investigacion y Desarrollo, S.A.U
+*
+* This file is part of Orion Context Broker.
+*
+* Orion Context Broker is free software: you can redistribute it and/or
+* modify it under the terms of the GNU Affero General Public License as
+* published by the Free Software Foundation, either version 3 of the
+* License, or (at your option) any later version.
+*
+* Orion Context Broker is distributed in the hope that it will be useful,
+* but WITHOUT ANY WARRANTY; without even the implied warranty of
+* MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU Affero
+* General Public License for more details.
+*
+* You should have received a copy of the GNU Affero General Public License
+* along with Orion Context Broker. If not, see http://www.gnu.org/licenses/.
+*
+* For those usages not covered by this license please contact with
+* iot_support at tid dot es
+*
+* Author: Ken Zangelin
+*/
+#include <string>
+#include <map>
+
+#include "logMsg/logMsg.h"
+#include "logMsg/traceLevels.h"
+
+#include "alarmMgr/AlarmManager.h"
+
+
+
+/* ****************************************************************************
+*
+* AlarmManager::AlarmManager - 
+*/
+AlarmManager::AlarmManager():
+dbOk(true),
+notificationErrorLogInterval(0),
+badInputLogInterval(0)
+{
+}
+
+
+
+/* ****************************************************************************
+*
+* AlarmManager::AlarmManager - 
+*/
+AlarmManager::AlarmManager(int _notificationErrorLogInterval, int _badInputLogInterval):
+dbOk(true),
+notificationErrorLogInterval(_notificationErrorLogInterval),
+badInputLogInterval(_badInputLogInterval)
+{
+}  
+
+
+
+/* ****************************************************************************
+*
+* AlarmManager::notificationErrorLogIntervalSet - 
+*/
+void AlarmManager::notificationErrorLogIntervalSet(int _notificationErrorLogInterval)
+{
+  notificationErrorLogInterval = _notificationErrorLogInterval;
+}
+
+
+
+/* ****************************************************************************
+*
+* AlarmManager::badInputLogIntervalSet - 
+*/
+void AlarmManager::badInputLogIntervalSet(int _badInputLogInterval)
+{
+  badInputLogInterval = _badInputLogInterval;
+}
+
+
+
+/* ****************************************************************************
+*
+* AlarmManager::dbError - 
+*
+* Returns false if no action is taken/necessary, otherwise, true is returned.
```

Comment:
I'd suggest something more descriptive than "no action", e.g.

"Returns false if no effective alarm transition (i.e. db was already alarmed) occur"

(Similar comment for adjustments in other cases in this PR)